liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -231,4 +219,25 @@ private String getSiblingModule(Artifact artifact) {
         }
         return libraries;
     }
+    
+    private File getWarSourceDirectory() {
+        // default location : ${basedir}/src/main/webapp
+        File dir = new File(project.getBasedir() + "/src/main/webapp");
+
+        @SuppressWarnings("unchecked")
+        List<Plugin> plugins = getProject().getBuildPlugins();
+        for (Plugin plugin : plugins) {
+            if ("org.apache.maven.plugins:maven-war-plugin".equals(plugin.getKey())) {
+                Object config = plugin.getConfiguration();
+                if (config instanceof Xpp3Dom) {
+                    Xpp3Dom dom = (Xpp3Dom) config;
+                    Xpp3Dom val = dom.getChild("warSourceDirectory");
+                    if (val != null) {
+                        return new File(val.getValue());
+                    }
+                }
```

Comment:
perhaps this can be refactored a bit to take the key and child element of interest as parameters to get the target/class setting as well.